pkg/controller/agent/reconcile.go

```diff
@@ -128,27 +131,38 @@ func reconcileDaemonSet(rp ReconciliationParams) (int32, int32, error) {
 	return reconciled.Status.NumberReady, reconciled.Status.DesiredNumberScheduled, nil
 }
 
+// ReconciliationParams are the parameters used during an Elastic Agent's reconciliation.
 type ReconciliationParams struct {
 	client      k8s.Client
 	agent       agentv1alpha1.Agent
 	podTemplate corev1.PodTemplateSpec
 }
 
-func updateStatus(params Params, ready, desired int32) error {
+// calculateStatus will calculate a new status from the state of the pods within the k8s cluster
+// and will return any error encountered.
+func calculateStatus(params *Params, ready, desired int32, status *agentv1alpha1.AgentStatus) error {
 	agent := params.Agent
 
 	pods, err := k8s.PodsMatchingLabels(params.Client, agent.Namespace, map[string]string{NameLabelName: agent.Name})
 	if err != nil {
 		return err
 	}
-	agent.Status.AvailableNodes = ready
-	agent.Status.ExpectedNodes = desired
+	status.AvailableNodes = ready
+	status.ExpectedNodes = desired
 	health, err := CalculateHealth(agent.GetAssociations(), ready, desired)
 	if err != nil {
 		return err
 	}
 	agent.Status.Health = health
```

Comment:
This update is lost IIUC. You are not updating `status`. Missing unit test?
```suggestion
	status.Health = health
```

Reply:
You were correct.  Unit test added for status.health.